### PR TITLE
Dubbo2.7.5: Add Page and Prioritized interface 

### PR DESCRIPTION
**Pull request by flycash**
In order to align with dubbo 2.7.5, we should implement the ServiceDiscovery which need this.

### CODE REVIEW

**Comment by AlexStocks** [comment]
@flycash pls add some comments to readme about this package.

**Comment by AlexStocks** [comment]
@flycash pls resolve the comment by @hound

File: page/dafault_page.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gxpage
+
+// DefaultPage is the default implementation of Page interface
+type DefaultPage struct {
```
Comment (wongoo) [inline page/dafault_page.go]: I wonder will user implement their own Page struct, 
- if yes, suggest rename interface `Page` to `Pager`, rename `DefaultPage` to `Page` which will be frequently, so the name should not keep simple
- if no,  remove the interface `Page`, rename `DefaultPage` to `Page`

**Comment by wongoo** [comment]
new review comment added

File: page/page.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ * limitations under the License.
+ */
+
+package gxpage
+
+// Page is the abstraction for pagination usage.
+type Page interface {
+
+	// GetOffSet will return the offset
+	GetOffset() int
```
Comment (zouyx) [inline page/page.go]: should  change to GetOffset ？

File: page/page.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	return d.hasNext
+}
+
+// HasData will return whether this page has data.
+func (d *Page) HasData() bool {
+	return d.GetDataSize() > 0
+}
+
+// NewDefaultPage will create an instance
+func NewDefaultPage(requestOffset int, pageSize int,
```
Comment (wongoo) [inline page/page.go]: rename to `New`

File: sort/prioritized.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gxsort
+
+// Prioritized is the abstraction of priority.
+type Prioritized interface {
```
Comment (wongoo) [inline sort/prioritized.go]: what's this interface use for? add some unit test for it.  And how about rename to `Prioritizer`

Comment (flycash) [reply sort/prioritized.go]: Used to declare some thing that should be ordered by its priority.
For example, the listener in dubbo2.7.5 should be ordered by priority. 
It's just a interface, we don't need UT.

File: page/page.go (hunk 3 above)
Comment (flycash) [reply page/page.go]: OK

**Comment by wongoo** [comment]
LGTM

File: page/page.go [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+type Page struct {
+	requestOffset int
+	pageSize      int
+	totalSize     int
+	data          []interface{}
+	totalPages    int
+	hasNext       bool
+}
+
+// GetOffSet will return the offset
```
Comment (zouyx) [inline page/page.go]: GetOffset not GetOffSet

**Pull request merged**